tests/chainer_tests/initializer_tests/test_constant.py

```diff
@@ -39,13 +41,21 @@ def check_initializer(self, w):
             **self.check_options)
 
     def test_initializer(self, backend_config):
-        w = backend_config.xp.empty(self.shape, dtype=self.dtype)
-        self.check_initializer(w)
+        if isinstance(backend_config.device,
+                      chainer.backends.intel64.Intel64Device):
+            pytest.skip("IDeep4py fails when using np.fill_diagonal")
 
-    def check_shaped_initializer(self, xp):
+        w = numpy.empty(self.shape, dtype=self.dtype)
+        w = backend_config.get_array(w)
+        with chainer.using_device(backend_config.device):
```

Comment:
`generate_array` should run without setting the current device.

Reply:
It fails when device is cuda_device 1